library/Rules/Email.php

```diff
@@ -5,6 +5,14 @@ class Email extends AbstractRule
 {
     public function validate($input)
     {
-        return is_string($input) && filter_var($input, FILTER_VALIDATE_EMAIL);
+        if (class_exists('\Egulias\EmailValidator\EmailValidator')) {
+            $strictValidator = new \Egulias\EmailValidator\EmailValidator();
```

Comment:
But you can't type hint, class may or may not exist.
El 26/2/2015 22:43, "Henrique Moody" <notifications@github.com> escribió:

> In library/Rules/Email.php
> <https://github.com/Respect/Validation/pull/299#discussion_r25467627>:
>
> > @@ -5,6 +5,14 @@ class Email extends AbstractRule
> >  {
> >      public function validate($input)
> >      {
> > -        return is_string($input) && filter_var($input, FILTER_VALIDATE_EMAIL);
> > +        if (class_exists('\Egulias\EmailValidator\EmailValidator')) {
> > +            $strictValidator = new \Egulias\EmailValidator\EmailValidator();
>
> PS: When testing use a mock of \Egulias\EmailValidator\EmailValidator and
> pass it as dependency to the rule instead of use the original class, them
> you can assert if Validation is making the correct call to EmailValidator.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Respect/Validation/pull/299/files#r25467627>.
>